apps/src/blockly/customBlocks/googleBlockly/spritelabBlocks.js

```diff
@@ -74,12 +77,19 @@ export const blocks = {
     }
 
     if (this.workspace.rendered) {
+      const imageSourceId = this.id;
       this.workspace.registerToolboxCategoryCallback(
-        `flyout_${this.type}`,
+        CdoFieldFlyout.getFlyoutId(this),
```

Comment:
It looks like `this.type` includes both `block.type` and `block.id`? (Just noticing that `getFlyoutId` returns more info than just `this.type`.)

Reply:
I changed this because with the previous logic (only using type) we could create two toolboxes with the same key (for example if you put two "when clicked" blocks into the workspace), and then I couldn't distinguish between the two workspaces for the purpose of image shadowing. So I made the keys unique per block. Does that answer your question or were you asking something else?

Reply:
That answers my question, thanks!